### ISSUE TITLE
ssl key password errors are confusing and cryptic

### ISSUE THREAD

**Issue by tcmitchell** [GENI-NSF/geni-tools#25]
If a user mistypes their ssl key password they get a lengthy stack trace that does not explain what went wrong.

 _Imported from trac ticket [#25](http://trac.gpolab.bbn.com/gcf/ticket/25),  created by tmitchel on 10-09-2010 at 08:38, last modified: 10-13-2010 at 15:50_
   CCing: ahelsing@...

**Comment by tcmitchell** [comment]
Here's a sample reported by a GEC 9 experimenter:
```
dark@dark-laptop:~/Desktop/gcf-1.1$ src/omni.py createsliver -a
http://myplc.gpolab.bbn.com:12346 -n netserv_gpo rspec/myplc-gpo.rspec
--debug
INFO:omni:Loading config file omni_config
INFO:omni:Using control framework pgeni
DEBUG:omni.protogeni:Configured with key file /home/dark/.ssl/gpo.pem
DEBUG:omni.protogeni:Using clearinghouse
https://www.pgeni.gpolab.bbn.com:443/protogeni/xmlrpc/ch
DEBUG:omni.protogeni:Using slice authority
https://www.pgeni.gpolab.bbn.com:443/protogeni/xmlrpc/sa
Enter PEM pass phrase:
DEBUG:omni.protogeni:Resolving
urn:publicid:IDN+pgeni.gpolab.bbn.com+slice+netserv_gpo at slice authority
Enter PEM pass phrase:
DEBUG:omni.protogeni:Got resolve response {'output': '', 'code': 0,
'value': {'creator_uuid': '2fa17c37-c03c-11df-afaf-000c29f89f7b', 'uuid':
'a12c1f0c-d2ea-11df-92ee-000c29f89f7b', 'urn':
'urn:publicid:IDN+pgeni.gpolab.bbn.com+slice+netserv_gpo',
'component_managers': [], 'hrn': 'bbn-pgeni.netserv_gpo', 'gid':
'MIIDgTCCAuoCAgCCMA0GCSqGSIb3DQEBBAUAMIHQMQswCQYDVQQGEwJVUzEWMBQG\nA1UECBMNTWFzc2FjaHVzZXR0czESMBAGA1UEBxMJQ2FtYnJpZGdlMRwwGgYDVQQK\nExNHRU5JIFByb2plY3QgT2ZmaWNlMR4wHAYDVQQLExVDZXJ0aWZpY2F0ZSBBdXRo\nb3JpdHkxIjAgBgNVBAMTGWJvc3MucGdlbmkuZ3BvbGFiLmJibi5jb20xMzAxBgkq\nhkiG9w0BCQEWJHRlc3RiZWQtb3BzQG9wcy5wZ2VuaS5ncG9sYWIuYmJuLmNvbTAe\nFw0xMDEwMDgxNDQ1MDFaFw0xNjAzMzAxNDQ1MDFaMIHHMQswCQYDVQQGEwJVUzEW\nMBQGA1UECBMNTWFzc2FjaHVzZXR0czEcMBoGA1UEChMTR0VOSSBQcm9qZWN0IE9m\nZmljZTEeMBwGA1UECxQVYmJuLXBnZW5pLm5ldHNlcnZfZ3BvMS0wKwYDVQQDEyRh\nMTJjMWYwYy1kMmVhLTExZGYtOTJlZS0wMDBjMjlmODlmN2IxMzAxBgkqhkiG9w0B\nCQEWJHJvYmVydG8uZnJhbmNlc2NhbmdlbGlAZGllaS51bmlwZy5pdDCBnzANBgkq\nhkiG9w0BAQEFAAOBjQAwgYkCgYEA5rChPZzRB/E3bvlkrapvTznbSZPOojzsbzTa\nh1bIFwswz3RkoyjzcRnS7LEbbQ301/4iVvo8sidjfYhgcjmZDkEGLC6GLR2Om2QO\nTOn+VnbyRzeNJt2HEabY8suvMKb119sH8FqZkpsVnbWgjyRRQY5u2HaE/0lJBGKG\n/kAylO8CAwEAAaN2MHQwDwYDVR0TAQH/BAUwAwEB/zAdBgNVHQ4EFgQUB6BFbLoM\nPGi/nwwNrLWzY4kZxbEwQgYDVR0RBDswOYY3dXJuOnB1YmxpY2lkOklETitwZ2Vu\
naS5ncG9sYWIuYmJuLmNvbStzbGljZStuZXRzZXJ2X2dwbzANBgkqhkiG9w0BAQQF\nAAOBgQCct+uI79GJQuRN7MAhNpkps6z7/woRNodT8Hdxh7xkxIbPR5Rwi7Fg8k+G\n/m7jBw/g5in7QcN2XTECLuKHtG1YrZWhlg2HQXf21FGyzXP1r53WgMg6eJC8Hs7H\nBY8mIpJ3sPmVCEMGFEE3dm+sW5bJ5c2KNwJfGlp1eyGtXHQtig==\n',
'creator_urn': 'urn:publicid:IDN+pgeni.gpolab.bbn.com+user+roberto'}}
DEBUG:omni.protogeni:Resolved slice
urn:publicid:IDN+pgeni.gpolab.bbn.com+slice+netserv_gpo, getting
credential
Enter PEM pass phrase:
Traceback (most recent call last):
 File "src/omni.py", line 613, in <module>
   sys.exit(main())
 File "src/omni.py", line 609, in main
   handler._handle(args)
 File "src/omni.py", line 113, in _handle
   getattr(self,call)(args[1:])
 File "src/omni.py", line 260, in createsliver
   slice_cred = self.framework.get_slice_cred(urn)
 File
"/home/dark/Desktop/gcf-1.1/src/omnilib/frameworks/framework_pg.py",
line 129, in get_slice_cred
   response = self.sa.GetCredential(params)
 File "/usr/lib/python2.6/xmlrpclib.py", line 1199, in __call__
   return self.__send(self.__name, args)
 File "/usr/lib/python2.6/xmlrpclib.py", line 1489, in __request
   verbose=self.__verbose
 File "/usr/lib/python2.6/xmlrpclib.py", line 1235, in request
   self.send_content(h, request_body)
 File "/usr/lib/python2.6/xmlrpclib.py", line 1349, in send_content
   connection.endheaders()
 File "/usr/lib/python2.6/httplib.py", line 904, in endheaders
   self._send_output()
 File "/usr/lib/python2.6/httplib.py", line 776, in _send_output
   self.send(msg)
 File "/usr/lib/python2.6/httplib.py", line 735, in send
   self.connect()
 File "/usr/lib/python2.6/httplib.py", line 1112, in connect
   self.sock = ssl.wrap_socket(sock, self.key_file, self.cert_file)
 File "/usr/lib/python2.6/ssl.py", line 350, in wrap_socket
   suppress_ragged_eofs=suppress_ragged_eofs)
 File "/usr/lib/python2.6/ssl.py", line 113, in __init__
   cert_reqs, ssl_version, ca_certs)
ssl.SSLError: [Errno 336265225] _ssl.c:337: error:140B0009:SSL
routines:SSL_CTX_use_PrivateKey_file:PEM lib
```

_Trac comment by tmitchel (github user: tcmitchell) on 10-09-2010 at 08:40_

**Comment by ahelsing** [comment]
SSL Key PW errors are now caught and the user gets to retry.

Note these shortcomings in this fix:
- there are other possible errors from the SSL module, equally ugly, that we do not yet catch
- the user may have to type this pass phrase multiple times
- we catch these errors at the top level, so the user may have to effectively repeat work (and type the pass phrase even more times)

This fix is available post gcf-1.1


_Trac comment by ahelsing on 10-13-2010 at 15:46_
